archaius-aws/src/main/java/com/netflix/config/sources/AbstractDynamoDbConfigurationSource.java

```diff
@@ -95,6 +106,25 @@ public AbstractDynamoDbConfigurationSource(AmazonDynamoDB dbClient) {
         this.dbClient = dbClient;
     }
 
+
+    protected ScanResult dbScanWithThroughputBackOff(ScanRequest scanRequest) {
+        Long currentBackOff = minBackOff.get();
+        while(true) {
+            try {
+                return dbClient.scan(scanRequest);
+            }
+            catch (ProvisionedThroughputExceededException e) {
+                currentBackOff = Math.min(currentBackOff * 2, maxBackOff.get());
+                log.info("Failed to poll Dynamo due to ProvisionedThroughputExceededException. Backing off.");
```

Comment:
Since this is an infinite loop I think at least a WARN if not an ERROR is justified here.